src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -141,9 +141,14 @@ def __call__(self, poller):
 # pylint: disable=too-few-public-methods
 class DeploymentOutputLongRunningOperation(LongRunningOperation):
     def __call__(self, poller):
-        result = super(DeploymentOutputLongRunningOperation, self).__call__(poller)
-        outputs = result.properties.outputs
-        return {key: val['value'] for key, val in outputs.items()} if outputs else {}
+        try:
+            result = super(DeploymentOutputLongRunningOperation, self).__call__(poller)
+            outputs = result.properties.outputs
+            return {key: val['value'] for key, val in outputs.items()} if outputs else {}
+        except AttributeError:
+            # in the event of an AttributeError, this is likely just the result
+            from msrest.pipeline import ClientRawResponse
+            return None if isinstance(poller, ClientRawResponse) else poller
```

Comment:
This isn't needed for --validate. It's needed for `--no-wait` to work correctly. The reason being that the DeploymentOutputLongRunningOperation expects a poller and calls the "done()" method, but this will not exist if you use --no-wait because you will get ClientRawResponse. 

Finally, if you used --validate you would get simply a response, not a poller and not a ClientRawResponse. Hence the need for this logic.